src/content/1.7/modules/sample_modules/order_view_page_new_hooks/view-order-hooks-signature-widget.md

```diff
@@ -0,0 +1,448 @@
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function getType()
+    {
+        return Order::class;
+    }
+}
+```
+
+`OrderRepository` in `src\Repository`:
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Repository;
+
+use DateTimeImmutable;
+use Db;
+use Order as PrestaShopOrder;
+use PrestaShop\Module\DemoViewOrderHooks\Collection\Orders;
+use PrestaShop\Module\DemoViewOrderHooks\DTO\Order;
+
+class OrderRepository
+{
+    /**
+     * @var Db
+     */
+    private $db;
+
+    public function __construct()
+    {
+        $this->db = Db::getInstance();
+    }
+
+    /**
+     * Get all orders that a customer has placed.
+     */
+    public function getCustomerOrders(int $customerId, array $excludeOrderIds = []): Orders
+    {
+        $orders = PrestaShopOrder::getCustomerOrders($customerId);
+        $ordersCollection = new Orders();
+
+        foreach ($orders as $order) {
+            if (in_array($order['id_order'], $excludeOrderIds)) {
+                continue;
+            }
+
+            $ordersCollection->add(new Order(
+                (int) $order['id_order'],
+                $order['reference'],
+                (int) $order['current_state'],
+                new DateTimeImmutable($order['date_add'])
+            ));
+        }
+
+        return $ordersCollection;
+    }
+}
+
+```
+Let's create `OrderSignaturePresenter` class responsible for returning order customer data  
+in `src/Presenter/`:
+
+```php
+<?php
+declare(strict_types=1);
+
+namespace PrestaShop\Module\DemoViewOrderHooks\Presenter;
+
+use Gender;
+use Order;
+use PrestaShop\Module\DemoViewOrderHooks\Entity\OrderSignature;
+
+class OrderSignaturePresenter
+{
+    /**
+     * @var string
+     */
+    private $signatureImgDir;
+
+    public function __construct(string $signatureImgDir)
+    {
+        $this->signatureImgDir = $signatureImgDir;
+    }
+
+    public function present(OrderSignature $signature, int $languageId): array
+    {
+        $order = new Order($signature->getOrderId());
+        $customer = $order->getCustomer();
+        $gender = new Gender($customer->id_gender, $languageId);
+
+        return [
+            'firstName' => $customer->firstname,
+            'lastName' => $customer->lastname,
+            'gender' => $gender->name,
+            'filename' => $this->signatureImgDir.$signature->getFilename()
+        ];
+    }
+}
+```
+Then lets use Symfony Dependency Injection 
+(https://www.freecodecamp.org/news/a-quick-intro-to-dependency-injection-what-it-is-and-when-to-use-it-7578c84fa88f/).
+and create services configuration for the above 
+classes in `config/services.yml`. For more information: 
+https://devdocs.prestashop.com/1.7/modules/concepts/services/#symfony-services
+The intent behind dependency injection is to achieve Separation of Concerns of construction and use of objects. 
+This can increase readability and code reuse, reduce dependencies, lead to more testable code.
```

Comment:
It also reduces memory consumption as services are, by default, created once and shared in the whole project.